### PR TITLE
fixed the mtls check for tls on sidecar

### PR DESCRIPTION
**Pull request by shriramsharma**
**Please provide a description of this PR:**
Quick fix for mtls check while building tls listener context for sidecar proxy. Related to this PR comment
https://github.com/istio/istio/pull/36210#discussion_r784420273 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ x] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
LGTM but a bit concerned no test cases were changed - do we need to increase our test coverage?

**Comment by shriramsharma** [comment]
> LGTM but a bit concerned no test cases were changed - do we need to increase our test coverage?

@howardjohn , I think you're right. I'll work on it.

File: releasenotes/notes/36912.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
```
Comment (bianpengyuan) [inline releasenotes/notes/36912.yaml:2]: I don't think we need a bug fix release notes for a not yet released feature?

Comment (bianpengyuan) [reply releasenotes/notes/36912.yaml]: Sorry I actually meant that we don't need a release note for this change at all. You can simply add `release-notes-none` label to the PR.

**Review by bianpengyuan — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**